src/hooks/usePosts.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+import { Post } from '../types';
+import * as postsService from '../api/posts';
+
+export const usePosts = (selectedUserId: number = 0) => {
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadSelectedUserPosts = () => {
+    if (!selectedUserId) {
+      return;
+    }
+
+    setIsLoading(true);
+    setErrorMessage('');
+
+    postsService
+      .getUserPosts(selectedUserId)
+      .then(setPosts)
+      .catch(() => setErrorMessage('Unable to load posts'))
```

Comment:
Consider providing more detailed error messages or logging the error details to help with debugging and improving user feedback.